packages/automerge-repo-network-websocket/src/BrowserWebSocketClientAdapter.ts

```diff
@@ -83,43 +96,19 @@ export class BrowserWebSocketClientAdapter extends WebSocketNetworkAdapter {
     if (!this.socket) {
       throw new Error("WTF, get a socket")
     }
-    this.socket.send(CBOR.encode({ type: "leave", senderId: this.peerId }))
+    this.transmit({ type: "leave", senderId: this.peerId! })
   }
 
-  sendMessage(
-    targetId: PeerId,
-    channelId: ChannelId,
-    message: Uint8Array,
-    broadcast: boolean
-  ) {
-    if (message.byteLength === 0) {
+  send(message: Message) {
+    if (message.data.byteLength === 0) {
       throw new Error("tried to send a zero-length message")
     }
+
     if (!this.peerId) {
       throw new Error("Why don't we have a PeerID?")
     }
 
-    const decoded: InboundMessagePayload = {
-      senderId: this.peerId,
-      targetId,
-      channelId,
-      type: "sync",
-      message,
-      broadcast,
-    }
-
-    const encoded = CBOR.encode(decoded)
-
-    // This incantation deals with websocket sending the whole
-    // underlying buffer even if we just have a uint8array view on it
-    const arrayBuf = encoded.buffer.slice(
-      encoded.byteOffset,
-      encoded.byteOffset + encoded.byteLength
-    )
-    if (!this.socket || this.socket.readyState !== WebSocket.OPEN) {
-      throw new Error("Websocket Socket not ready!")
-    }
-    this.socket.send(arrayBuf)
+    this.transmit(message)
```

Comment:
is there a reason to not just make "send()" send and have the other functions use it?

Reply:
(presumably we don't want to send zero length messages or send without a peerId inside transmit either)

Reply:
Will look at that today. Transmit also accepts internal message types, not just ones which match the Adapter interface signature.

This method also changed name from `sendMessage` to `send` while I was migrating the code. There is no reason for this other than it made it easier for me, so can be changed back to `sendMessage` straightforwardly should you prefer.

Reply:
This is now done, and tests passing.

would you like it named back to `sendMessage`?